qemu/tests/passthrough_with_multipath_device.py

```diff
@@ -0,0 +1,86 @@
+import logging
+import time
+import random
+import re
+
+from avocado.utils import process
+
+from virttest import error_context
+from virttest import env_process
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test failover path on SCSI passthrough with underlying DM-multipath device.
+
+    Step:
+     1. Build multipath device on host.
+     2. Boot a guest with passthrough path.
+     3. Access guest then do io on the data disk.
+     4. Check vm status.
+     5. Alternately close a path every 10 seconds on host
+     6. Check vm status
+     7. Offline two path and check status
+     8. Online one path and check status
+
+
+    :param test: Qemu test object.
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+
+    vm = env.get_vm(params['main_vm'])
+    timeout = float(params.get("login_timeout", 240))
+    get_id_cmd = params.get("get_id_cmd")
+    get_mpath_cmd = params.get("get_mpath_cmd")
+    get_mdev_cmd = params.get("get_mdev_cmd")
+    set_path_cmd = params.get("set_path_cmd")
+    post_cmd = params.get("post_cmd")
+    repeat_time = params.get_numeric("repeat_time")
+    id = process.getoutput(get_id_cmd).strip()
+    get_mpath_cmd = get_mpath_cmd % (id, id)
+    mpath = process.getoutput(get_mpath_cmd).strip()
+    params["image_name_stg0"] = "/dev/mapper/%s" % mpath
+    params['start_vm'] = 'yes'
+    env_process.process(test, params, env, env_process.preprocess_image,
+                        env_process.preprocess_vm)
+    session = vm.wait_for_login(timeout=timeout)
+    cmd_dd = "dd if=/dev/zero of=/dev/sdb bs=1M count=30000 &"
```

Comment:
Do you test by manually it have enough time run the io, you may try 
dd if=/dev/zero of=/dev/sdb bs=4k count=1000000 iflag=direct